### PR TITLE
Initial Leafish Flatpak

### PR DESCRIPTION
**Pull request by TheEvilSkeleton**
Closes https://github.com/terrarier2111/Leafish/issues/104.

Since this MR requires approval, test CIs will be available at https://github.com/TheEvilSkeleton/Leafish/pull/1.

### CODE REVIEW

**Comment by PureTryOut** [comment]
I like it so far! CI is failing though.

I'm wondering, does this CI workflow push the eventual Flatpak to a repo of some sorts people can use?

**Comment by PureTryOut** [comment]
Uh nvm, CI isn't failing, I was looking at an old commit :see_no_evil: 

So yeah just that question, otherwise this looks good to go.

File: packaging/flatpak/com.github.terrarier2111.Leafish.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+rename-icon: leafish-logo
+rename-desktop-file: leafish.desktop
+rename-appdata-file: leafish.metainfo.xml
+finish-args:
+- "--share=ipc"
+- "--share=network"
+- "--socket=fallback-x11"
+- "--socket=wayland"
+- "--device=dri"
+- "--socket=pulseaudio"
```
Comment (PureTryOut) [inline packaging/flatpak/com.github.terrarier2111.Leafish.yaml:17]: We don't currently have audio yet, maybe remove this for now till we do?

**Comment by TheEvilSkeleton** [comment]
> I'm wondering, does this CI workflow push the eventual Flatpak to a repo of some sorts people can use?

Sadly not. We need to create a Flatpak remote for that. For now, we can only manually install using artifacts.

File: packaging/flatpak/com.github.terrarier2111.Leafish.yaml (hunk 1 above)
Comment (TheEvilSkeleton) [reply packaging/flatpak/com.github.terrarier2111.Leafish.yaml:17]:
```suggestion
```

Comment (TheEvilSkeleton) [reply packaging/flatpak/com.github.terrarier2111.Leafish.yaml:17]: > We don't currently have audio yet, maybe remove this for now till we do?

If you want to, I can remove it.

Comment (PureTryOut) [reply packaging/flatpak/com.github.terrarier2111.Leafish.yaml:17]: Please do for now, easy enough to add back later. Let's keep permissions only to what's necessary

**Comment by PureTryOut** [comment]
@TheEvilSkeleton how did you generate `cargo-sources.json`? I wanted to update a dependency but CI now fails because of it not being present in that file :thinking:

**Comment by TheEvilSkeleton** [comment]
@PureTryOut this should help you out: https://github.com/flatpak/flatpak-builder-tools/tree/master/cargo

If you need help, feel free to ask.

**Comment by PureTryOut** [comment]
Interesting. I'm not sure I'm a fan of developers having to install those tools somehow locally and running it every time they want to update some dependency :thinking: 

> The intended usage of the generators is as a submodule used as part of your build process to generate manifests.

Can we not copy the script to our repo and run it as part of some `cargo build` or `cargo run` command or something?

**Comment by TheEvilSkeleton** [comment]
> Interesting. I'm not sure I'm a fan of developers having to install those tools somehow locally and running it every time they want to update some dependency thinking

It's not really mandatory, but advisable. Now that everyone pushes reproducible builds, we kind of have to do that. If you wan, we can give the build system network access, but I wouldn't advise it.

> Can we not ~copy~ include the script as a submodule and run it as part of some `cargo build` or `cargo run` command or something?

Of course we can :)

I don't know how to include a single script as a submodule, though. I only know how to use repos as submodules.

**Comment by PureTryOut** [comment]
Well you can include the repo as a submodule but then only use the script from it. If that's doable then please add that, it would make it so much easier to change and update deps in the future!

**Pull request merged**